pkgs/development/python-modules/causal-conv1d/default.nix

```
@@ -0,0 +1,61 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchFromGitHub,
+  ninja,
+  torch,
+  cudaPackages,
+  rocmPackages,
+  config,
+  cudaSupport ? config.cudaSupport,
+  which,
+}:
+
+buildPythonPackage rec {
+  pname = "causal-conv1d";
+  version = "1.4.0";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "Dao-AILab";
+    repo = pname;
+    rev = "refs/tags/v${version}";
+    hash = "sha256-p5x5u3zEmEMN3mWd88o3jmcpKUnovTvn7I9jIOj/ie0=";
+  };
+
+  build-system = [
+    ninja
+    which
+  ];
```

```suggestion
  build-system = [
    ninja
    setuptools
    torch
  ];
  
  nativeBuildInputs = [ which ];
```

`build-system` only accepts python libraries.

https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#buildpythonpackage-parameters-buildpythonpackage-parameters

---

interestingly enough, moving `which` to `buildInputs` does not work, neither does omitting `which` here.

---

We are in the migration process and there is no difference in implementation between `build-system` and `nativeBuildInputs` for compatibility.
Thanks for your understanding.